src/main/java/nextstep/subway/application/LineService.java

```diff
@@ -85,7 +85,20 @@ public SectionCreateResponse createSection(long id, SectionCreateRequest request
                 upDownStations.findById(downStationId),
                 new Distance(request.getDistance()), line));
 
-        return SectionCreateResponse.of(line, line.getSectionLineUpInOrder());
+        return LineSectionResponse.of(line, line.getSectionLineUpInOrder());
+    }
+
+    @Transactional(isolation = READ_COMMITTED)
+    public LineSectionResponse deleteSection(long lineId, long stationId) {
+        Line line = lineRepository.findById(lineId)
+                .orElseThrow(() -> new IllegalArgumentException("요청한 노선을 찾을 수 없습니다. 노선ID:" + lineId));
+        Station station = stationService.findById(stationId);
+        List<Section> deletedSections = line.deleteSection(station);
+        sectionRepository.deleteAllById(deletedSections.stream()
+                .map(Section::getId)
+                .collect(Collectors.toList()));
```

Comment:
```suggestion
        line.deleteSection(station);
```
`orphanRemoval` 옵션을 사용하면 명시적으로 삭제요청을 하지 않더라도 삭제할수 있어요! 😀
어떤 경우에 `orphanRemoval`을 사용하면 좋을지 고민해보면 좋을것같네요.

[JPA 고아객체](https://tgyun615.com/185)

Reply:
초기에 `노선`과 `구간`의 연관관계를 설정할 때, `orphanRemoval` 옵션을 `true`로 주었기에 
이미 `hibernate`에서 `flush` 할 때 연관관계가 끊어진 `구간`들에 대해서 `delete` 쿼리를 발생시키고 있었는데
`hibernate`에서 `flush`하는 타이밍을 제가 잘못 이해하고 있었네요 !
`부모` 객체의 컬렉션에서 제거되면 고아가 된 `자식`객체들은 제거되는 것은 맞는데,
다만 그것을 영속성 컨텍스트에서  갖고 있다가 `flush`가 발생하는 타이밍에 함께 발생하는 구조인데
디버깅 하는 시점에 `delete` 쿼리가 나가지 않아서 `orphanRemoval` 조건에 걸리지 않는다고 착각한 것 같아요 ! 😭
다시 디버깅 해본 결과, `부모` 객체에서 제거 되어 더이상 `부모`에 속하지 않는 `자식` 객체들을 제거하는 `orphanRemoval` 옵션 동작 확인하여서
명시적으로 `구간`을 삭제해주던 부분은 제거하였습니다 !
자칫 놓치고 넘어갈 수 있었던 부분이었는데, 꼼꼼하게 확인해주셔서 감사합니다 !! ☺️👍